### PR TITLE
[Spi flash] testutils refactor to return status_t

### PR DESCRIPTION
**Pull request by engdoreis**
See https://github.com/lowRISC/opentitan/issues/17495

### CODE REVIEW

File: sw/device/tests/sim_dv/spi_passthrough_test.c [hunk 1]
```diff
@@ -235,7 +235,7 @@ void handle_sector_erase(void) {
       dif_spi_device_get_4b_address_mode(&spi_device, &addr4b_enabled));
 
   bool addr_is_4b = dif_toggle_to_bool(addr4b_enabled);
-  spi_flash_testutils_erase_sector(&spi_host0, address, addr_is_4b);
+  TRY(spi_flash_testutils_erase_sector(&spi_host0, address, addr_is_4b));
```
Comment (cfrantz) [inline sw/device/tests/sim_dv/spi_passthrough_test.c:238]: CHECK_STATUS_OK ?

**Review by cfrantz — APPROVED** [review]

File: sw/device/tests/sim_dv/spi_passthrough_test.c (hunk 1 above)
Comment (engdoreis) [reply sw/device/tests/sim_dv/spi_passthrough_test.c:238]: Good catch, fixed.

File: sw/device/lib/testing/spi_flash_testutils.c [hunk 2]
```diff
@@ -196,26 +200,28 @@ void spi_flash_testutils_erase_op(dif_spi_host_t *spih, uint8_t opcode,
...
 
-void spi_flash_testutils_erase_sector(dif_spi_host_t *spih, uint32_t address,
-                                      bool addr_is_4b) {
-  spi_flash_testutils_erase_op(spih, kSpiDeviceFlashOpSectorErase, address,
-                               addr_is_4b);
+status_t spi_flash_testutils_erase_sector(dif_spi_host_t *spih,
+                                          uint32_t address, bool addr_is_4b) {
+  TRY(spi_flash_testutils_erase_op(spih, kSpiDeviceFlashOpSectorErase, address,
+                                   addr_is_4b));
+  return OK_STATUS();
```
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/device/lib/testing/spi_flash_testutils.c:214]: My regex is not smart enough, Fixed!

File: sw/device/lib/testing/spi_flash_testutils.c [hunk 3]
```diff
@@ -243,27 +249,31 @@ void spi_flash_testutils_program_op(dif_spi_host_t *spih, uint8_t opcode,
...
-                                      bool addr_is_4b) {
-  spi_flash_testutils_program_op(spih, kSpiDeviceFlashOpPageProgram, payload,
-                                 length, address, addr_is_4b);
+status_t spi_flash_testutils_program_page(dif_spi_host_t *spih,
+                                          const void *payload, size_t length,
+                                          uint32_t address, bool addr_is_4b) {
+  TRY(spi_flash_testutils_program_op(spih, kSpiDeviceFlashOpPageProgram,
+                                     payload, length, address, addr_is_4b));
+
+  return OK_STATUS();
```
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/device/lib/testing/spi_flash_testutils.c]: Fixed

**Pull request merged**